src/server/scripts/Spells/spell_paladin.cpp

```diff
@@ -1430,34 +1032,55 @@ class spell_pal_t8_2p_bonus : public SpellScriptLoader
         }
 };
 
+// 269569 - Zeal
+class spell_pal_zeal : public AuraScript
+{
+    PrepareAuraScript(spell_pal_zeal);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_PALADIN_JUDGEMENT_PROT_RET_R3, SPELL_PALADIN_JUDGEMENT_GAIN_HOLY_POWER });
+    }
+
+    void HandleEffectProc(AuraEffect* aurEff, ProcEventInfo& /*procInfo*/)
+    {
+        Unit* caster = GetCaster();
```

Comment:
caster may be null

Reply:
GetTarget() is what you wanna use for auras that equal getcaster in 9/10 cases

Reply:
Yeah I actually fixed that already after @mdX7 comment. Realized I should of used GetTarget + corrected the validate method.